src/librustc_mir/transform/mod.rs

```diff
@@ -155,17 +155,30 @@ pub trait MirPass {
                           mir: &mut Mir<'tcx>);
 }
 
-pub macro run_passes($tcx:ident, $mir:ident, $def_id:ident, $suite_index:expr; $($pass:expr,)*) {{
-    let suite_index: usize = $suite_index;
+pub macro run_passes(
```

Comment:
Unrelated: it'd be nice to move this out of a macro and into some kind of generic function. I don't really see any inherent difficulty in that. The functon could take a `vec![&dyn MirPass]` or else we could do some hancy h-list sort of thing (hint: seems unnecessary =)